src/pages/RFOX/components/Bridge/BridgeConfirm.tsx

```diff
@@ -0,0 +1,416 @@
+    if (!quote || quote.isErr()) return
+
+    const tradeQuote = quote.unwrap()
+
+    return tradeQuote.steps[0].allowanceContract
+  }, [quote])
+
+  const { data: allowanceData, isLoading: isAllowanceDataLoading } = useAllowance({
+    assetId: bridgeQuote.sellAssetId,
+    spender: allowanceContract,
+    from: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+  })
+
+  const isApprovalRequired = useMemo(
+    () => bnOrZero(allowanceData).lt(bridgeQuote.bridgeAmountCryptoBaseUnit),
+    [allowanceData, bridgeQuote.bridgeAmountCryptoBaseUnit],
+  )
+
+  const approvalCallData = useMemo(() => {
+    if (!allowanceContract) return
+
+    return encodeFunctionData({
+      abi: erc20ABI,
+      functionName: 'approve',
+      args: [getAddress(allowanceContract), BigInt(bridgeQuote.bridgeAmountCryptoBaseUnit)],
+    })
+  }, [allowanceContract, bridgeQuote.bridgeAmountCryptoBaseUnit])
+
+  const {
+    mutate: sendApprovalTx,
+    isPending: isApprovalMutationPending,
+    isSuccess: isApprovalMutationSuccess,
+    isIdle: isApprovalMutationIdle,
+  } = useMutation({
+    ...reactQueries.mutations.approve({
+      assetId: bridgeQuote.sellAssetId,
+      spender: allowanceContract!,
+      from: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+      amount: bridgeQuote.bridgeAmountCryptoBaseUnit,
+      wallet,
+      accountNumber,
+    }),
+    onSuccess: (txHash: string) => {
+      setApprovalTxHash(txHash)
+      toast({
+        title: translate('modals.send.transactionSent'),
+        description: (
+          <Text>
+            {feeAsset?.explorerTxLink && (
+              <Link href={`${feeAsset.explorerTxLink}${txHash}`} isExternal>
+                {translate('modals.status.viewExplorer')} <ExternalLinkIcon mx='2px' />
+              </Link>
+            )}
+          </Text>
+        ),
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+        position: 'top-right',
+      })
+    },
+  })
+
+  const isGetApprovalFeesEnabled = useMemo(
+    () =>
+      Boolean(
+        isApprovalMutationIdle &&
+          isApprovalRequired &&
+          feeAsset &&
+          feeAssetMarketData &&
+          wallet &&
+          accountNumber !== undefined,
+      ),
+    [
+      accountNumber,
+      feeAsset,
+      feeAssetMarketData,
+      isApprovalMutationIdle,
+      isApprovalRequired,
+      wallet,
+    ],
+  )
+
+  const { data: approvalFees, isLoading: isGetApprovalFeesLoading } = useQuery({
+    ...reactQueries.common.evmFees({
+      value: '0',
+      accountNumber: accountNumber!, // see isGetApprovalFeesEnabled
+      feeAsset: feeAsset!, // see isGetApprovalFeesEnabled
+      feeAssetMarketData: feeAssetMarketData!, // see isGetApprovalFeesEnabled
+      to: fromAssetId(bridgeQuote.sellAssetId).assetReference,
+      from: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+      data: approvalCallData!,
+      wallet: wallet!, // see isGetApprovalFeesEnabled
```

Comment:
this isn't nice 💀 is there a less assert-y way of solving this?

Reply:
would be possible by wrapping the component and call this query in a context where these are defined

Reply:
See discussion here: https://discord.com/channels/554694662431178782/1248445558260109322

Will bump react-query in the p3 (cleanup) PR while at it to leverage `skipToken` safety

Reply:
Tackled in p3: https://github.com/shapeshift/web/pull/7079/commits/ff065afd95f7fa1f0aac4aab3b4798e274e2a53a and https://github.com/shapeshift/web/pull/7079/commits/bfdb5a54caa666cd3d33555b1cb90916b79b6cb6

This isn't exhaustive obviously, as tackling all would take days, but serving as an example of how we can do proper type narrowing.

The second is probably the best way to do this, but is obviously an additional layer of abstraction we may not always need - it allows us to have our cake and eat it to clearly separate 
- programmatic query enabling/disabling i.e enabling the query depending on a criteria, but *not* including type narrowing which is a diff concern
- type narrowing separately, so the programmatic enabling is obvious (and if there is no programmatic enabling, then it's even more obvious we're only doing type narrowing)

Note the second example can be done at component-level as well, which is what the first does - the only reason why it doesn't do both is because it doesn't require programmatic enabling